src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1879,6 +1885,73 @@ private BoundExpression BindCast(CastExpressionSyntax node, DiagnosticBag diagno
             return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics);
         }
 
+        private BoundExpression BindIndexExpression(PrefixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            CheckFeatureAvailability(node, MessageID.IDS_FeatureIndexOperator, diagnostics);
+            GetWellKnownTypeMember(Compilation, WellKnownMember.System_Index__ctor, diagnostics, syntax: node);
+
+            return BindUnaryOperator(node, diagnostics);
+        }
+
+        private BoundExpression BindRangeExpression(RangeExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            CheckFeatureAvailability(node, MessageID.IDS_FeatureRangeOperator, diagnostics);
+            TypeSymbol rangeType = GetWellKnownType(WellKnownType.System_Range, diagnostics, node);
+
+            if (!rangeType.IsErrorType())
+            {
+                WellKnownMember requiredRangeMethod;
+
+                if (node.LeftOperand is null)
+                {
+                    requiredRangeMethod = node.RightOperand is null ? WellKnownMember.System_Range__All : WellKnownMember.System_Range__ToEnd;
+                }
+                else
+                {
+                    requiredRangeMethod = node.RightOperand is null ? WellKnownMember.System_Range__FromStart : WellKnownMember.System_Range__Create;
+                }
+
+                GetWellKnownTypeMember(Compilation, requiredRangeMethod, diagnostics, syntax: node);
```

Comment:
>GetWellKnownTypeMember [](start = 16, length = 22)

I think we should save the method in the bound node for the benefit of GetSymbolInfo and IOperation APIs

Reply:
added a prototype comment

---
In reply to: [185304525](https://github.com/dotnet/roslyn/pull/26405#discussion_r185304525) [](ancestors = 185304525)